Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -0,0 +1,387 @@
+            set { SetValue(ItemRotationYProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for Rotation.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty ItemRotationYProperty = DependencyProperty.Register("ItemRotationY", typeof(double), typeof(Carousel), new PropertyMetadata(0d, OnCarouselPropertyChanged));
+
+        /// <summary>
+        /// Gets or sets rotation angle on Z
+        /// </summary>
+        public double ItemRotationZ
+        {
+            get { return (double)GetValue(ItemRotationZProperty); }
+            set { SetValue(ItemRotationZProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for Rotation.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty ItemRotationZProperty = DependencyProperty.Register("ItemRotationZ", typeof(double), typeof(Carousel), new PropertyMetadata(0d, OnCarouselPropertyChanged));
+
+        /// <summary>
+        /// Gets or sets the Carousel orientation. Horizontal or Vertical
+        /// </summary>
+        public Orientation Orientation
+        {
+            get { return (Orientation)GetValue(OrientationProperty); }
+            set { SetValue(OrientationProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for MaxViewableItems.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty OrientationProperty = DependencyProperty.Register("Orientation", typeof(Orientation), typeof(Carousel), new PropertyMetadata(Orientation.Horizontal, OnCarouselPropertyChanged));
+
+        /// <summary>
+        /// Each time a property change, we have to reposition
+        /// </summary>
+        private static void OnCarouselPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            if (e.NewValue == null)
+            {
+                return;
+            }
+
+            Carousel carouselControl = (Carousel)d;
+            if (carouselControl.ItemsPanel == null)
+            {
+                return;
+            }
+
+            var itemsPanel = carouselControl.GetItemsPanel();
+
+            if (itemsPanel == null)
+            {
+                return;
+            }
+
+            itemsPanel.UpdatePosition();
+        }
+
+        /// <summary>
+        /// Return ItemsPanel
+        /// </summary>
+        /// <returns>Gets the ItemsPanel used for the carousel</returns>
+        internal CarouselPanel GetItemsPanel()
+        {
+            return this.FindDescendant<CarouselPanel>();
+        }
+
+        /// <summary>
+        /// Returns the container used for each item
+        /// </summary>
+        /// <returns>Returns always a ContentControl</returns>
+        protected override DependencyObject GetContainerForItemOverride()
```

Comment:
It doesn't work without specifying that item must be a contentcontrol.
If we delete this, the PrepareContainerForItemOverride is failing.